khal/khalendar/event.py

```diff
@@ -342,6 +342,40 @@ def summary(self):
     def update_summary(self, summary):
         self._vevents[self.ref]['SUMMARY'] = summary
 
+    @staticmethod
+    def _can_handle_alarm(alarm):
+        """
+        Decides whether we can handle a certain alarm.
+        """
+        return alarm.get('ACTION') == 'DISPLAY' and isinstance(alarm.get('TRIGGER').dt, timedelta)
+
+    @property
+    def alarms(self):
+        """
+        Returns a list of all alarms in th original event that we can handle. Unknown types of
+        alarms are ignored.
+        """
+        return [(a.get('TRIGGER').dt, a.get('DESCRIPTION'))
+                for a in self._vevents[self.ref].subcomponents
+                if a.name == 'VALARM' and self._can_handle_alarm(a)]
+
+    def updated_alarms(self, alarms):
```

Comment:
You can use the `alarms.setter` decorator here, FWIW, but better to stay consistent.

But: please rename to `update_alarms`